CRM/Utils/Mail.php

```diff
@@ -577,4 +577,25 @@ public static function format($fields) {
     return $formattedEmail;
   }
 
+  /**
+   * When passed a value, returns the value if it's non-numeric.
+   * If it's numeric, look up the display name and email of the corresponding
+   * contact ID in RFC822 format.
+   *
+   * @param string $header
+   * @return string
+   *   The RFC822-formatted email header (display name + address)
+   */
+  public static function mailboxHeader($header) {
+    if (is_numeric($header)) {
+      $result = civicrm_api3('Email', 'get', [
+        'id' => $header,
+        'return' => ['contact_id.display_name', 'email'],
+        'sequential' => 1,
+      ])['values'][0];
```

Comment:
This will give a PHP notice if the id is not found

Reply:
That's probably a good thing though, right?  The only legitimate use case in which this code path executes would require the id to be found.  Or are you suggesting I should use `getsingle` so it throws an error rather than a notice?

Reply:
The old and this new code pretends that ```is_numeric($from)``` will always be a valid email ID, so it doesn't need any additional handling for empty result. Or I could have suggested:
```
$result = CRM_Utils_Array::value(0, civicrm_api3('Email', 'get', [
        'id' => $from,
        'return' => ['contact_id.display_name', 'email'],
        'sequential' => 1,
      ])['values']);
 if (empty($result)) {
  Civi::log()->warning(ts('Invalid From email ID - ' . $from);
}
...
```
I think we can ignore this.